dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/McpTransport.java

```diff
@@ -271,4 +277,37 @@ public String getRequestIpAddress() {
     public McpProtocolVersion getProtocolVersion() {
         return version;
     }
+
+    public <T> CompletionStage<Void> sendResultAsync(CompletionStage<T> stage) {
+        AsyncContext asyncContext = req.startAsync();
+        asyncContext.setTimeout(TimeUnit.SECONDS.toMillis(TIMEOUT_SECONDS.get()));
+
+        return stage.handle((result, throwable) -> {
+            try {
+                if (throwable == null) {
+                    try {
+                        sendResponse(result);
+                    } catch (Exception io) {
+                        Tr.error(tc, "CWMCM0016E.error.sending.response.exception", io);
+                        try {
+                            sendError(new Exception(io));
+                        } catch (IOException e) {
+                            Tr.error(tc, "CWMCM0016E.error.sending.response.exception", e);
+                        }
+                    }
+                } else {
+                    try {
+                        sendError(throwable);
+                    } catch (IOException e) {
+                        Tr.error(tc, "CWMCM0016E.error.sending.response.exception", e);
+                    }
+                }
+            } catch (Exception e) {
+                sendResponse(ToolResponse.error(Tr.formatMessage(tc, "CWMCM0011E.internal.server.error")));
+            } finally {
+                asyncContext.complete();
+            }
+            return null;
+        });
```

Comment:
There are a lot of cases here. Would this be sufficient instead?

```suggestion
        return stage.handle((result, throwable) -> {
            try {
                if (throwable == null) {
                    sendResponse(result);
                } else {
                    sendError(throwable);
                }
            } catch (Exception e) {
                // There was a problem sending the response
                // This could be a network issue (in which case it's probably not recoverable) or some kind of serialization issue
                // Try sending this error (which will also log it), but if there's another problem, just ignore it
                try {
                    sendError(e);
                } catch (Exception e2) {
                }
            } finally {
                asyncContext.complete();
            }
            return null;
        });
```

Reply:
done